src/Hosting/Hosting/src/GenericHost/GenericWebHostService.cs

```diff
@@ -73,6 +73,18 @@ public async Task StartAsync(CancellationToken cancellationToken)
                 urls = Options.WebHostOptions.ServerUrls;
             }
 
+            if (string.IsNullOrEmpty(urls))
+            {
+                // HTTP_PORTS and HTTPS_PORTS, these are lower priority than Urls.
+                var httpPorts = Configuration[WebHostDefaults.HttpPortKey];
+                var httpUrls = httpPorts?.Split(';', StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries)
+                    .Select(port => $"http://*:{port}").Aggregate((s1, s2) => string.Concat(s1, ";", s2));
```

Comment:
nit: Put this code into a local method, have the key and prefix as args and then reuse for HTTP and HTTPS